lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -100,13 +104,13 @@ def success?(response)
 
   def error_message(response)
     explanations = {
-      '400' => 'Something is wrong with the data sent to Foreman server',
+      '400' => "Something is wrong with the data sent to #{fqdn}",
       '401' => 'Often this is caused by invalid Oauth credentials',
       '404' => 'The requested resource was not found',
-      '500' => 'Check /var/log/foreman/production.log on Foreman server for detailed information',
-      '502' => 'The webserver received an invalid response from the backend service. Was Foreman unable to handle the request?',
-      '503' => 'The webserver was unable to reach the backend service. Is foreman.service running?',
-      '504' => 'The webserver timed out waiting for a response from the backend service. Is Foreman under unusually heavy load?'
+      '500' => "Check /var/log/foreman/production.log on #{fqdn} for detailed information",
+      '502' => "The webserver received an invalid response from the backend service. Was #{fqdn} unable to handle the request?",
```

Comment:
I think in this case we refer to the service Foreman, not the server. I'm not sure this statement still makes sense.

Reply:
I will change this to `"The webserver received an invalid response from the backend service. Was foreman.service at #{fqdn} unable to handle the request?"`